src/averageAges.js

```diff
@@ -54,8 +68,20 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  let children = people
+    .filter(child => people
+      .find(mother => child.mother === mother.name));
```

Comment:
you can use method `some` to make your code more readable
Because method `find` returns the element or `undefined` and others developers, when looking at your code, expect you to do something with the element found